test/e2e/framework/route_client.go

```diff
@@ -64,7 +64,10 @@ func NewRouteClient(
 	var token string
 
 	for _, secret := range secrets.Items {
-		if strings.Contains(secret.Name, "cluster-monitoring-operator-e2e-token-") {
+		_, dockerToken := secret.Annotations["openshift.io/create-dockercfg-secrets"]
+		e2eToken := strings.Contains(secret.Name, "cluster-monitoring-operator-e2e-token-")
+
+		if !dockerToken && e2eToken {
```

Comment:
Can we just add a comment that if its dockerToken that means this token is an invalid one, to avoid future confusion.

Reply:
sounds good